src/main/java/uk/gov/hmcts/ccd/data/casedetails/CachedCaseDetailsRepository.java

```diff
@@ -43,29 +44,52 @@ public CaseDetails set(final CaseDetails caseDetails) {
         return caseDetailsRepository.set(caseDetails);
     }
 
+    @Override
+    public Optional<CaseDetails> findById(String jurisdiction, Long id) {
+        final Function<Long, CaseDetails> findCase = (key) -> caseDetailsRepository.findById(jurisdiction, id)
+                                                                                   .orElse(null);
+        return Optional.ofNullable(idToCaseDetails.computeIfAbsent(id, findCase))
+                       .filter(caseDetails -> jurisdiction.equals(caseDetails.getJurisdiction()));
+    }
+
     @Override
     public CaseDetails findById(final Long id) {
         return idToCaseDetails.computeIfAbsent(id, caseDetailsRepository::findById);
     }
 
     @Override
     public CaseDetails findByReference(final Long caseReference) {
-        return referenceToCaseDetails.computeIfAbsent(caseReference, caseDetailsRepository::findByReference);
+        return referenceToCaseDetails.computeIfAbsent(caseReference.toString(), (key) -> caseDetailsRepository.findByReference(caseReference));
+    }
+
+    @Override
+    public Optional<CaseDetails> findByReference(String jurisdiction, Long reference) {
+        return findByReference(jurisdiction, reference.toString());
     }
 
     @Override
-    public Optional<CaseDetails> findByReference(String jurisdictionId, Long caseReference) {
-        if (referenceToCaseDetails.containsKey(caseReference)) {
-            return Optional.ofNullable(referenceToCaseDetails.get(caseReference))
-                           .filter(caseDetails -> jurisdictionId.equals(caseDetails.getJurisdiction()));
+    public Optional<CaseDetails> findByReference(String jurisdiction, String reference) {
+        if (referenceToCaseDetails.containsKey(reference)) {
+            return Optional.ofNullable(referenceToCaseDetails.get(reference))
+                           .filter(caseDetails -> jurisdiction.equals(caseDetails.getJurisdiction()));
```

Comment:
because in case of cache miss we are doing a get by jurisdiction and id, is this check necessary?
Also, do we have to check this at all given that case reference is unique across jurisdictions?